addons/account/tests/test_account_move_out_invoice.py

```diff
@@ -3446,13 +3446,15 @@ def test_change_currency_id(self):
                 'display_type': 'product',
                 'currency_id': self.currency_data['currency'].id,
                 'debit': 0.0,
-                'credit': 375.0,
+                'credit': 750.0,
+                'amount_currency': -1500.0,
```

Comment:
The amount in currency must be the same as the `price_unit`

Reply:
Indeed this not seems correct

Reply:
So?

Reply:
Hello, sorry for the delay.
I worked a bit more on this and another solution is to avoid updating `amount_currency` before the record is actually saved.
To recap current issue with the tax line (not sure why it does not affect product line):
- we start with an invoice in foreign currency, let's say `amount_currency` is 450, `balance` is 150, `currency_rate` is 3.0
- switch to company currency -> `amount currency = balance = 150`
- switch to foreign currency -> no update to amount currency
- save -> `balance = amount_currency / currency_rate = 150 / 3.0 = 50`